File: sources/net.sf.j2s.ajax/ajaxpipeserlvet/net/sf/j2s/ajax/SimplePipeRPCHttpServlet.java
/*******************************************************************************
 * Copyright (c) 2007 java2script.org and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     Zhou Renjian - initial API and implementation
 *******************************************************************************/
package net.sf.j2s.ajax;

import net.sf.j2s.ajax.SimpleRPCHttpServlet;
import net.sf.j2s.ajax.SimpleRPCRunnable;
import net.sf.j2s.ajax.SimpleSerializable;

/**
 * 
 * @author zhou renjian
 */
public class SimplePipeRPCHttpServlet extends SimpleRPCHttpServlet {

	private static final long serialVersionUID = 3141977053834098092L;

	@Override
	protected SimpleRPCRunnable getRunnableByRequest(String request) {
		SimpleRPCRunnable runnable = super.getRunnableByRequest(request);
		if (runnable instanceof SimplePipeRunnable) {
			SimplePipeRunnable pipeRunnable = (SimplePipeRunnable) runnable;
			pipeRunnable.setPipeHelper(new SimplePipeHelper.IPipeThrough() {
			
				public void helpThrough(SimplePipeRunnable pipe,
						SimpleSerializable[] objs) {
					SimplePipeHelper.pipeIn(pipe.pipeKey, objs);
				}
			
			});
			pipeRunnable.pipeManaged = managingPipe;
		}
		return runnable;
	}

}

File: old/incubator/net.sf.j2s.test.mauve/j2slib/net/sf/j2s/ajax/SimplePipeRPCHttpServlet.js
$_J("net.sf.j2s.ajax");
$_L(["net.sf.j2s.ajax.SimpleRPCHttpServlet"],"net.sf.j2s.ajax.SimplePipeRPCHttpServlet",["net.sf.j2s.ajax.SimplePipeHelper","net.sf.j2s.ajax.SimplePipeHelper.IPipeThrough"],function(){
c$=$_T(net.sf.j2s.ajax,"SimplePipeRPCHttpServlet",net.sf.j2s.ajax.SimpleRPCHttpServlet);
$_M(c$,"getRunnableByRequest",
function(request){
var runnable=$_U(this,net.sf.j2s.ajax.SimplePipeRPCHttpServlet,"getRunnableByRequest",[request]);
if($_O(runnable,net.sf.j2s.ajax.SimplePipeRunnable)){
var pipeRunnable=runnable;
pipeRunnable.setPipeHelper((($_D("net.sf.j2s.ajax.SimplePipeRPCHttpServlet$1")?0:net.sf.j2s.ajax.SimplePipeRPCHttpServlet.$SimplePipeRPCHttpServlet$1$()),$_N(net.sf.j2s.ajax.SimplePipeRPCHttpServlet$1,this,null)));
pipeRunnable.pipeManaged=this.managingPipe;
}return runnable;
},"~S");
c$.$SimplePipeRPCHttpServlet$1$=function(){
$_H();
c$=$_W(net.sf.j2s.ajax,"SimplePipeRPCHttpServlet$1",null,net.sf.j2s.ajax.SimplePipeHelper.IPipeThrough);
$_V(c$,"helpThrough",
function(pipe,objs){
net.sf.j2s.ajax.SimplePipeHelper.pipeIn(pipe.pipeKey,objs);
},"net.sf.j2s.ajax.SimplePipeRunnable,~A");
c$=$_P();
};
});
